crates/db_views/src/structs.rs

```diff
@@ -237,3 +243,49 @@ pub struct LocalImageView {
   pub local_image: LocalImage,
   pub person: Person,
 }
+
+#[derive(Debug, PartialEq, Serialize, Deserialize, Clone)]
+#[cfg_attr(feature = "full", derive(Queryable))]
+#[cfg_attr(feature = "full", diesel(check_for_backend(diesel::pg::Pg)))]
+/// A combined report view
+pub struct ReportCombinedViewInternal {
+  // Post-specific
+  pub post_report: Option<PostReport>,
+  pub post: Option<Post>,
+  pub post_counts: Option<PostAggregates>,
+  pub post_unread_comments: Option<i64>,
+  pub post_saved: bool,
+  pub post_read: bool,
+  pub post_hidden: bool,
+  pub my_post_vote: Option<i16>,
+  // Comment-specific
+  pub comment_report: Option<CommentReport>,
+  pub comment: Option<Comment>,
+  pub comment_counts: Option<CommentAggregates>,
+  pub comment_saved: bool,
+  pub my_comment_vote: Option<i16>,
+  // Private-message-specific
+  pub private_message_report: Option<PrivateMessageReport>,
+  pub private_message: Option<PrivateMessage>,
+  // Shared
+  pub report_creator: Person,
+  pub item_creator: Person,
+  pub community: Option<Community>,
+  pub subscribed: SubscribedType,
+  pub resolver: Option<Person>,
+  pub item_creator_is_admin: bool,
+  pub item_creator_banned_from_community: bool,
+  pub item_creator_is_moderator: bool,
+  pub item_creator_blocked: bool,
+}
```

Comment:
Can you simplify this similar to `ModlogCombinedViewInternal` in the other PR, or does that not work with the db query?

```rust
pub struct ReportCombinedViewInternal {
  pub cost: Option<PostReportView>,
  pub comment: Option<CommentReportView>,
  pub private_message<PrivateMessageReportView>,
}
```

Reply:
That can't work for either one. The *Internal* is the actual DB query that selects everything necessary, then a mapping function maps that data to the enum, which is a container for the different types.